src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
-      'https://maps.googleapis.com/maps/api/staticmap' +
-      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng + 
-      '&zoom=18&size=500x500&key=' + apiKey)
-    });
-  let geometry = new THREE.PlaneGeometry(50,50);
+        'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + median.lat + ',' + median.lng +
+      '&zoom=18&size=500x500&key=' + apiKey),
+  });
+  const geometry = new THREE.PlaneGeometry(200, 200);
   const map = new THREE.Mesh(geometry, material);
-  map.position.set(0,-4,0);
-  map.rotation.x = -Math.PI / 2;
+  map.position.set(0, 0, -4);
   scene.add(map);
 
   // Add the light to the scene.
-  let light = new THREE.PointLight(0xffffff, 1, 0 );
-  light.position.set(0, 100, 0);
-  scene.add(light)
+  const light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 0, 1000);
+  scene.add(light);
+
+  createInstances();
   plotTrajectory();
-  
-  // Adds pose data to the scene as one Geometry instance.
-  material = new THREE.MeshBasicMaterial({color: 'red'});
-  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
-  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
-  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
-  scene.add(orientation);
   plotOrientation();
 }
 
+
 /**
- * This uses the pose data to create a blue line representing the pose
- * trajectory. 
+ * Creates all the children for each axis of orientation. Each object
+ * is a geometry instance, initializing the object with one geometry
+ * call rather than the number of data points. (O(1) vs. O(N))
+ */
+function createInstances() {
+  // Blue corresponds to the Z axis.
+  let material = new THREE.MeshBasicMaterial({color: 'blue'});
+  const geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation.Z = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Z.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Z);
+
+  // Green corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'green'});
+  orientation.Y = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.Y.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.Y);
+
+  // Red corresponds to the Z axis.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  orientation.X = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.X.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation.X);
+}
+/**
+  * Converts latitude, longitude, and altitude coordinates (LLA) to
+  * Earth-Centered, Earth-Fixed coordinates (ECEF).
+  * @param {double} lat Latitude represented in Decimal Degrees
+  * @param {double} lng Longitude represented in Decimal Degrees
+  * @param {double} alt Altitude represented in Decimal Degrees
+  * @return {array} An array containing x, y, z ECEF coordinates.
+  */
+function llaToEcef(lat, lng, alt) {
+  // Constants.
```

Comment:
Include units.

Reply:
Do you mean units for the constants in the function? I wrote the lat, long, and alt coordinates were in decimal degrees above.

Reply:
Yes, for the constants (those which aren't dimensionless, of course).

Reply:
Oh, and for the return values.